SU2_CFD/src/solvers/CEulerSolver.cpp

```diff
@@ -5764,9 +5860,189 @@ void CEulerSolver::SetActDisk_BCThrust(CGeometry *geometry, CSolver **solver_con
+                  text_line_appo.erase (0,5);
+                  istringstream axis_value(text_line_appo);
+                  for (iDim = 0; iDim < nDim; iDim++){
+                     axis_value >> AD_Axis[iDim];
+                  }
+
+                  /*--- Read and assign the value of the actuator disk radius. ---*/
+                  getline (ActDisk_file, text_line_appo);
+                  text_line_appo.erase (0,7);
+                  istringstream R_value(text_line_appo);
+                  R_value >> AD_Radius;
+
+                  /*--- Read and assign the value of the actuator disk advance ratio. ---*/
+                  getline (ActDisk_file, text_line_appo);
+                  text_line_appo.erase (0,10);
+                  istringstream J_value(text_line_appo);
+                  J_value >> AD_J;
+
+                  /*--- Read and assign the number of radial stations contained in the propeller data file. ---*/
+                  getline (ActDisk_file, text_line_appo);
+                  text_line_appo.erase (0,5);
+                  istringstream row_value(text_line_appo);
+                  row_value >> nRow;
+
+                  /*--- Read and assign the values of the non-dimensional radius, thrust coefficient, power coefficient
+                        and radial force coefficient. ---*/
+                  getline (ActDisk_file, text_line_appo);
+                  for (iRow=0; iRow < nRow; iRow++){
+                     getline (ActDisk_file, text_line_appo);
+                     istringstream row_val_value(text_line_appo);
+                     row_val_value >> rad_v[iRow] >> dCt_v[iRow] >> dCp_v[iRow] >> dCr_v[iRow];
+                  }
+
+                  /*--- Set the actuator disk radius value, center coordiantes values and axis coordinates values. ---*/
+                  SetActDisk_R(iMarker, AD_Radius);
+                  for (iDim = 0; iDim < nDim; iDim++){
+                     SetActDisk_C(iMarker, iDim, AD_Center[iDim]);
+                     SetActDisk_Axis(iMarker, iDim, AD_Axis[iDim]);
+                  }
+
+                  /*--- If the first radial station corresponds to the actuator disk center, the radial and tangential forces
+                        per unit area (Fr and Ft) are equal to zero, while the axial force per unit area (Fa) is computed using
+                        a linear interpolation in order to avoid a mathematical singularity at actuator disk center. ---*/
+                  if (rad_v[0] == 0.0){
+                      Fa[0] = (((2*Dens_FreeStream*pow(Vel_FreeStream[0],2))/
+                              (pow(AD_J,2)*PI_NUMBER))*((dCt_v[1] - dCt_v[0])/rad_v[1])) / config->GetPressure_Ref();
+                      Ft[0] = 0.0;
+                      Fr[0] = 0.0;
+                  }
+                  else{
+                    Fa[0] = (dCt_v[0]*(2*Dens_FreeStream*pow(Vel_FreeStream[0],2))/
+                            (pow(AD_J,2)*PI_NUMBER*rad_v[0])) / config->GetPressure_Ref();
```

Comment:
:+1: thanks to me it reads more naturally like this.

Reply:
Yes, it is more clear like this!